src/ScratchStorage.js

```diff
@@ -171,6 +183,27 @@ class ScratchStorage {
             tryNextHelper();
         });
     }
+
+    /**
+     * Store an asset by type & ID.
+     * @param {AssetType} assetType - The type of asset to fetch. This also determines which asset store to use.
+     * @param {DataFormat} [dataFormat] - Optional: load this format instead of the AssetType's default.
+     * @param {DataFormat} [data] - Optional: store this data, rather than the data in the asset
+     * @param {?string} assetId - The ID of the asset to fetch: a project ID, MD5, etc.
+     * @return {Promise.<id, title>} A promise for asset metadata
+     */
+    store (assetType, dataFormat, data, assetId) {
+        dataFormat = dataFormat || assetType.runtimeFormat;
+        return new Promise(
+            (resolve, reject) =>
+                this.webHelper.store(assetType, dataFormat, data, assetId)
+                    .then(body => {
+                        this.builtinHelper.store(assetType, dataFormat, data, body.id);
+                        return resolve(body);
```

Comment:
Is the `return` unnecessary here?

Reply:
By convention you should `return` when you call the callback so that you don't accidentally do stuff after it (like calling the callback a second time). Not strictly necessary in terms of code though.